luarules/gadgets/cus_gl4.lua

```diff
@@ -945,6 +956,8 @@ local function initBinsAndTextures()
 				objectDefToUniformBin[unitDefID] = 'armunit'
 			elseif 	unitDef.name:sub(1,3) == 'cor' then
 				objectDefToUniformBin[unitDefID] = 'corunit'
+			elseif 	unitDef.name:sub(1,3) == 'ion' then -- temporary prefix for legion during transition from cor/arm to legion atlas
```

Comment:
@WatchTheFort will have a seizure on the floor when he sees you add more name parsing and it's not even using an obvious prefix

Reply:
The prefix was choosen by ender in the discord, coming from the second half of legion, leg-ION.
There will be plenty of legion units for now still using the cortex atlas, unless you want to go and rename them all to cor_legwhatevers.
Using a new prefix should also be a better alternative to manually adding each new unit in, and we have quite a few units that are now looking to use the new atlas.

Reply:
It seems that faction is something fundamental to each unit, so we should bite the bullet and add it as a custom param to every unit def. Then we won't need any more of this string parsing nonsense.

Reply:
Have two customparams, one for raw faction, and one for atlas. Have atlas default to faction so it only needs to be set explicitly for units who use a different faction's atlas.

Reply:
i don't know how two params would work for that, made it so if there is an atlas custom param it tries to use that, otherwise the first 3 letters.

Reply:
```
faction = faction or autogenerated()
atlas = atlas or faction
```

Reply:
i think i overengineered to that specification?